logger.go

```diff
@@ -38,3 +41,14 @@ func Must(zaplog *zap.Logger, err error) *zap.Logger {
 
 	return zaplog
 }
+
+// TestNew calls New, but returns both the logger, and an observer that can be
+// used to fetch and compare delivered logs.
+func TestNew(tb testing.TB, options ...zap.Option) (*zap.Logger, *observer.ObservedLogs) {
```

Comment:
I'd expect `NewTest`, if all the "constructor" functions begin with `New` it helps with autocompletion etc

Reply:
I prefer `TestNew`, as this is supposed to a "public testing API" as described by Mitchell Hashimoto.

See:

* https://speakerdeck.com/mitchellh/advanced-testing-with-go?slide=52
* https://www.youtube.com/watch?v=yszygk1cpEc

Right now, it's only a single function, but there could be more down the line, and then it makes sense to always prefix them with `Test` (and move them into `testing.go`, but I didn't see the point of that for this one function, but maybe I should).

Also, in terms of autocompletion, all editors I use actually use fuzzy matching, so not sure if that's a reason to change this default.